src/wallet/bitcoin.ts

```diff
@@ -15,152 +17,153 @@ export interface BitcoinWallet {
-    network: string,
-    peerUri: string,
-    hdKey: string
-  ): Promise<InMemoryBitcoinWallet> {
-    const parsedNetwork = Network.get(network);
-
-    const logger = new Logger({
-      level: "warning"
-    });
-    const walletdb = new WalletDB({
-      memory: true,
-      witness: true,
-      logger,
-      network: parsedNetwork
-    });
-    const chain = new Chain({
-      spv: true,
-      logger,
-      network: parsedNetwork
-    });
-    const pool = new Pool({
-      chain,
-      logger
-    });
-
-    await logger.open();
-    await pool.open();
-    await walletdb.open();
-    await chain.open();
-    await pool.connect();
-
-    const wallet = await walletdb.create({
-      logger,
-      network: parsedNetwork,
-      master: hdKey
-    });
-
-    const account = await wallet.getAccount(0);
-
-    for (let i = 0; i < 100; i++) {
-      pool.watchAddress(await account.deriveReceive(i).getAddress());
-      pool.watchAddress(await account.deriveChange(i).getAddress());
+    httpUrl: string,
+    username: string,
+    password: string,
+    network: Network,
+    hdKey?: string,
+    rescan?: boolean
+  ): Promise<BitcoindWallet> {
+    const auth = { username, password };
+
+    const key = !!hdKey
+      ? bip32.fromBase58(hdKey)
+      : bip32.fromSeed(crypto.randomBytes(32), networks[network]);
+
+    // use HD key fingerprint as wallet name
+    const walletName = key.fingerprint.toString("hex");
+    const walletExists = await axios
+      .post(
+        httpUrl,
+        {
+          jsonrpc: "1.0",
+          method: "listwallets"
+        },
+        { auth }
+      )
+      .then(res => res.data.result.includes(walletName));
+
+    if (!walletExists) {
+      await axios.post(
+        httpUrl,
+        {
+          jsonrpc: "1.0",
+          method: "createwallet",
+          params: [walletName]
+        },
+        { auth }
+      );
     }
 
-    pool.startSync();
-
-    pool.on("tx", (tx: any) => {
-      walletdb.addTX(tx);
-    });
-
-    pool.on("block", (block: any) => {
-      walletdb.addBlock(block);
-      if (block.txs.length > 0) {
-        block.txs.forEach((tx: any) => {
-          walletdb.addTX(tx);
-        });
+    let descriptor = `wpkh(${key.toBase58()}/0h/0h/*h)`;
+    const descriptorInfo = await axios
+      .post(
+        httpUrl,
+        {
+          jsonrpc: "1.0",
+          method: "getdescriptorinfo",
+          params: [descriptor]
+        },
+        { auth }
+      )
+      .then(res => res.data.result);
+    descriptor = `${descriptor}#${descriptorInfo.checksum}`;
+
+    const rpcRequestConfig = {
+      url: `${httpUrl}/wallet/${walletName}`,
+      method: "post" as Method,
+      auth
+    };
+
+    await axios.request({
+      ...rpcRequestConfig,
+      data: {
+        jsonrpc: "1.0",
+        method: "importmulti",
+        params: [
+          [{ desc: descriptor, timestamp: 0, range: 0 }],
+          { rescan: !!rescan ? rescan : false }
```

Comment:
instead of having this complicated logic here, I think you could just default to false in the parameter :)